src/js/helpers.js

```diff
@@ -769,14 +833,80 @@ export default class Helpers {
       $editPanel.toggle()
     }
     this.updatePreview($(field))
-    if (field.classList.contains('editing')) {
-      this.formBuilder.currentEditPanel = $editPanel[0]
-      config.opts.onOpenFieldEdit($editPanel[0])
-      document.dispatchEvent(events.fieldEditOpened)
+
+    const prevHolder = $(field).find('.prev-holder')
+    const fieldID = prevHolder.attr('data-field-id')
+    const liContainer = $(`#${fieldID}`)
+    const rowContainer = $(`#${fieldID}-cont`)
+
+    //Put the li back in its place
+    rowContainer.append(liContainer)
+
+    config.opts.onCloseFieldEdit($editPanel[0])
+    document.dispatchEvent(events.fieldEditClosed)
+    this.toggleGridModeButtonVisible(fieldID)
+
+    setTimeout(() => {
```

Comment:
be sure to clear timeouts to prevent memory leaks

```suggestion
    const resultsTimeout = setTimeout(() => {
      clearTimeout(resultsTimeout)
```